helm/customer-managed-agent/templates/raw_deployment.yaml

```diff
@@ -0,0 +1,131 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ include "helm.fullname" . }}-deployment-agent
+  labels:
+  {{- include "helm.labels" . | nindent 4 }}
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  - pods/log
+  - configmaps
+  verbs:
+  - create
+  - get
+  - list
+  - watch
+  - update
+  - delete
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: {{ include "helm.fullname" . }}-deployment-agent
+  labels:
+  {{- include "helm.labels" . | nindent 4 }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: '{{ include "helm.fullname" . }}-deployment-agent'
+subjects:
+- kind: ServiceAccount
+  name: '{{ include "helm.fullname" . }}-deployment-agent'
+  namespace: '{{ .Release.Namespace }}'
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "helm.fullname" . }}-pool
+  labels:
+  {{- include "helm.labels" . | nindent 4 }}
+  annotations:
+    app.kubernetes.io/name: pulumi-deployment-agent-pool
+spec:
+  replicas: {{ .Values.pool.replicas }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: customer-managed-deployment-agent
+    {{- include "helm.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: customer-managed-deployment-agent
+      {{- include "helm.selectorLabels" . | nindent 8 }}
+    spec:
+      containers:
+      - env:
+        - name: PULUMI_AGENT_DEPLOY_TARGET
+          value: kubernetes
+        - name: PULUMI_AGENT_SHARED_VOLUME_DIRECTORY
+          value: {{ quote .Values.pool.agent.env.pulumiAgentSharedVolumeDirectory }}
+        - name: PULUMI_AGENT_SERVICE_URL
+          valueFrom:
+            configMapKeyRef:
+              key: PULUMI_AGENT_SERVICE_URL
+              name: {{ include "helm.fullname" . }}-agent-config
+        - name: PULUMI_AGENT_IMAGE
+          valueFrom:
+            configMapKeyRef:
+              key: PULUMI_AGENT_IMAGE
+              name: {{ include "helm.fullname" . }}-agent-config
+        - name: PULUMI_AGENT_IMAGE_PULL_POLICY
+          valueFrom:
+            configMapKeyRef:
+              key: PULUMI_AGENT_IMAGE_PULL_POLICY
+              name: {{ include "helm.fullname" . }}-agent-config
+        - name: PULUMI_AGENT_TOKEN
+          valueFrom:
+            secretKeyRef:
+              key: PULUMI_AGENT_TOKEN
+              name: {{ include "helm.fullname" . }}-agent-secret
+        - name: KUBERNETES_CLUSTER_DOMAIN
+          value: {{ quote .Values.kubernetesClusterDomain }}
+        image: {{ .Values.pool.agent.image.repository }}:{{ .Values.pool.agent.image.tag
+          | default .Chart.AppVersion }}
+        imagePullPolicy: {{ .Values.pool.agent.imagePullPolicy }}
+        name: agent
+        resources: {}
+        volumeMounts:
+        - mountPath: /mnt/work
+          name: agent-work
+      serviceAccountName: {{ include "helm.fullname" . }}-deployment-agent
+      volumes:
+      - emptyDir: {}
```

Comment:
Why do we hardcode this? Is there some data users might want to persist?

Should we instead default to an emptyDir but allow folks to override this?

Reply:
I don't know why it's in there anyway, I've just taken it from here: https://github.com/pulumi/customer-managed-deployment-agent/blob/6bfb9bdd201a0f669ad34aa4313d6d1e66b9852b/kubernetes/agent.ts#L205-L208